pkg/conversion/deep_equal_test.go

```diff
@@ -50,14 +50,24 @@ func TestEqualities(t *testing.T) {
 	}{
 		{1, 2, true},
 		{2, 1, false},
+		{"foo", "fo", false},
 		{"foo", "foo", true},
+		{"foo", "foobar", false},
 		{Foo{1}, Foo{2}, true},
+		{Foo{2}, Foo{1}, false},
 		{Bar{1}, Bar{10}, true},
 		{&Bar{1}, &Bar{10}, true},
 		{Baz{Bar{1}}, Baz{Bar{10}}, true},
+		{map[string]int{"foo": 1}, map[string]int{}, false},
 		{map[string]int{"foo": 1}, map[string]int{"foo": 2}, true},
+		{map[string]int{"foo": 2}, map[string]int{"foo": 1}, false},
+		{map[string]int{"foo": 1}, map[string]int{"foo": 2, "bar": 6}, false},
+		{map[string]int{"foo": 1, "bar": 6}, map[string]int{"foo": 2}, false},
 		{map[string]int{}, map[string]int(nil), true},
-		{[]int{}, []int(nil), true},
+		{[]string{}, []string(nil), true},
```

Comment:
Please add: {[]string{"a"}, []string(nil), false},

Also please add some tests for arrays (e.g., `[3]int{1,2,3}`), since you changed that code.